contest/contest_info.py

```diff
@@ -99,6 +103,72 @@ def get_scoreboard(contest):
 
     return scoreboard
 
+def get_scoreboard_csv(contest_id):
+    try:
+        contest = Contest.objects.get(pk = contest_id)
+    except Contest.DoesNotExist:
+        logger.warning('Contest: Can not download Contest scoreboard %s! Contest not found!' % contest_id)
+        raise Http404('Contest does not exist, can not download scoreboard.')
+    scoreboard = get_scoreboard(contest)
+    
+    response = HttpResponse(content_type='text/csv')
+    filename = str(contest.cname) + "-scoreboard"
+    response['Content-Disposition'] = 'attachment; filename=' + filename
+
+    #init
+    writer = csv.writer(response)
+    #penalty scoreboard csv
+    scoreboard.sort_users_by_penalty()
+    #title
+    title = ['Penalty','User']
+    for problem in scoreboard.problems:
+        title.append(problem.id)
+    title.append('Total')
+    writer.writerow(title)
+    #user data
+    for counter, user in enumerate(scoreboard.users):
+        user_row = [counter, user.username]
+        for problem in user.problems:
+            penalty = problem.get_penalty(contest.start_time)
+            user_row.append(penalty)
```

Comment:
I prefer using another format:
```submissions before first AC``` / ```first AC time```(in minutes)
if a problem didn't get AC, use ```--``` instead
For example,
if an user got AC for the first submission, and submission time is 10 minutes after contest start, use ```1/10```
if an user had tried 4 times but never got AC, use ```4/--```